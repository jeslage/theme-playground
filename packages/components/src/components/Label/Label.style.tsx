import styled from 'styled-components';

const StyledLabel = styled.div`
  flex-grow: 2;
  padding-right: 20px;

  .label {
    margin: 0;
    display: flex;
    font-size: 14px;
    align-items: center;

    svg {
      width: 20px;
      height: auto;
      margin-right: 15px;
      fill: ${({ theme }) => theme.border};
    }
  }

  .description {
    margin: 0;
    margin-top: 10px;
  }
`;

export default StyledLabel;
