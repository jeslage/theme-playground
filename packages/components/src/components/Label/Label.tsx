import * as React from 'react';

import StyledLabel from './Label.style';

export interface Props {
  iconBefore?: JSX.Element;
  label?: string;
  description?: string;
}

const Label: React.FC<Props> = ({ label, iconBefore, description }) => (
  <StyledLabel>
    {(label || iconBefore) && (
      <p className="label">
        {iconBefore}
        {label && label}
      </p>
    )}
    {description && (
      <p className="description">
        <small>{description}</small>
      </p>
    )}
  </StyledLabel>
);

export default Label;
