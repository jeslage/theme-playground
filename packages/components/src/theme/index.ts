export const darkTheme = {
  base: 'dark',
  backgroundDark: '#2f2f2f',
  backgroundLight: '#333',
  border: '#999999',
  text: '#FFFFFF'
};

export const lightTheme = {
  base: 'light',
  backgroundDark: '#f6f9fc',
  backgroundLight: '#fff',
  border: '#666666',
  text: '#333333'
};

export const getUiTheme = theme => {
  if (theme === undefined) return darkTheme;

  if (typeof theme === 'string') {
    if (theme === 'dark') {
      return darkTheme;
    } else if (theme === 'light') {
      return lightTheme;
    }

    throw Error(
      `Theme "${theme}" is not available. Please use "light" or "dark", or provide a custom theme object.`
    );
  }

  if (theme.base && theme.base === 'light') {
    return {
      ...lightTheme,
      ...theme
    };
  }

  return {
    ...darkTheme,
    ...theme
  };
};
