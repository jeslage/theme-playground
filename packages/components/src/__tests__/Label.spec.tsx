import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import Label from '../components/Label/Label';
import StyledLabel from '../components/Label/Label.style';
import { darkTheme } from '../theme';

describe('Label', () => {
  it('Label should render correctly', () => {
    const component = shallow(
      <Label
        label="Label"
        description="Testbeschreibung"
        iconBefore={
          <svg height="24" viewBox="0 0 24 24" width="24">
            <rect height="2" rx="1" width="12" x="6" y="11" />
          </svg>
        }
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('StyledLabel should render correctly', () => {
    const component = shallow(<StyledLabel theme={{ colors: darkTheme }} />);

    expect(component).toMatchSnapshot();
  });
});
