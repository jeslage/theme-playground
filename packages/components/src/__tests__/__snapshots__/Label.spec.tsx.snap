// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Label should render correctly 1`] = `
<styled.div>
  <p
    className="label"
  >
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <rect
        height="2"
        rx="1"
        width="12"
        x="6"
        y="11"
      />
    </svg>
    Label
  </p>
  <p
    className="description"
  >
    <small>
      Testbeschreibung
    </small>
  </p>
</styled.div>
`;

exports[`Label StyledLabel should render correctly 1`] = `
<div
  className="sc-AxjAm kexaai"
/>
`;
